backend/app/db/dataset/version.py

```diff
@@ -165,3 +173,298 @@ async def _freeze_dataset_visualization(
+        ) is None:
+            fs.remove_object(settings.MINIO_BUCKET_NAME, str(frozen_file.origin_id))
+
+    return frozen_file.dict()
+
+
+async def _delete_frozen_dataset(frozen_dataset, fs, hard_delete=False):
+    # delete metadata
+    await MetadataFreezeDB.find(
+        MetadataFreezeDB.resource.resource_id == PydanticObjectId(frozen_dataset.id),
+        MetadataFreezeDB.resource.collection == "datasets",
+    ).delete()
+
+    # delete folders
+    await FolderFreezeDB.find(
+        FolderFreezeDB.dataset_id == PydanticObjectId(frozen_dataset.id)
+    ).delete()
+
+    # delete dataset thumbnails
+    await _delete_frozen_thumbnail(frozen_dataset, fs)
+
+    # delete dataset visualization
+    await _delete_frozen_visualizations(frozen_dataset, fs)
+
+    # delete files and file associate resources
+    async for frozen_file in FileFreezeDB.find(
+        FileFreezeDB.dataset_id == PydanticObjectId(frozen_dataset.id)
+    ):
+        if frozen_file.storage_type == StorageType.LOCAL:
+            await remove_frozen_local_file_entry(frozen_file.id)
+        else:
+            await remove_frozen_file_entry(frozen_file.id, fs)
+
+    # delete authorizations
+    await AuthorizationDB.find(
+        AuthorizationDB.dataset_id == PydanticObjectId(frozen_dataset.id)
+    ).delete()
+
+    # If all above succeeded
+    if hard_delete:
+        await frozen_dataset.delete()
+    else:
+        frozen_dataset.deleted = True
+        await frozen_dataset.save()
+
+    return frozen_dataset.dict()
+
+
+# TODO: Move this to MongoDB middle layer
+async def remove_file_entry(
+    file_id: Union[str, ObjectId],
+    fs: Minio,
+    es: Elasticsearch,
+):
+    """Remove FileDB object into MongoDB, Minio, and associated metadata and version information."""
+
+    if (file := await FileDB.get(PydanticObjectId(file_id))) is not None:
+        # delete from elasticsearch
+        delete_document_by_id(es, settings.elasticsearch_index, str(file_id))
+
+        # TODO: Deleting individual versions will require updating version_id in mongo, or deleting entire document
+        await FileVersionDB.find(FileVersionDB.file_id == ObjectId(file_id)).delete()
+
+        # delete metadata
+        await MetadataDB.find(
+            MetadataDB.resource.resource_id == ObjectId(file_id),
+            MetadataDB.resource.collection == "files",
+        ).delete()
+
+        # delete thumbnail
+        await _delete_thumbnail(file, fs)
+
+        # delete visualization
+        await _delete_visualizations(file, fs)
+
+        # delete file raw bytes if not used anywhere else
+        await _delete_file(file, fs)
+
+    else:
+        raise HTTPException(status_code=404, detail=f"File {file_id} not found")
+
+
+async def remove_frozen_file_entry(frozne_file_id: Union[str, ObjectId], fs: Minio):
+    """Remove Frozen FileDB object into MongoDB, Minio, and associated metadata and version information."""
```

Comment:
Are you deleting the file from the Minio here in this function?

Reply:
Yes same as above, I'm deleting them in here
```
await _delete_frozen_file(frozen_file, fs)
```